osu.Game/Rulesets/UI/HitObjectContainer.cs

```diff
@@ -178,6 +182,26 @@ protected override void OnChildLifetimeBoundaryCrossed(LifetimeBoundaryCrossedEv
 
         #endregion
 
+        /// <summary>
+        /// Invoked when a <see cref="DrawableHitObject"/> is added to this container.
+        /// </summary>
+        /// <remarks>
+        /// This method is not invoked for nested <see cref="DrawableHitObject"/>s.
+        /// </remarks>
+        protected virtual void OnAdd(DrawableHitObject drawableHitObject)
+        {
+        }
+
+        /// <summary>
+        /// Invoked when a <see cref="DrawableHitObject"/> is removed from this container.
+        /// </summary>
+        /// <remarks>
+        /// This method is not invoked for nested <see cref="DrawableHitObject"/>s.
```

Comment:
I think this may become an issue with the editor, but let's go with this for now and reconsider if that comes up.